File: project/yeonwoo-onboarding/src/main/java/com/innercircle/yeonwoo_onboarding/controller/SurveyController.java
package com.innercircle.yeonwoo_onboarding.controller;

import com.innercircle.yeonwoo_onboarding.domain.Survey;
import com.innercircle.yeonwoo_onboarding.service.SurveyService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/surveys")
@RequiredArgsConstructor
public class SurveyController {
    private final SurveyService surveyService;

    @GetMapping
    public ResponseEntity<List<Survey>> getAllSurveys() {
        return ResponseEntity.ok(surveyService.findAllSurveys());
    }

    @GetMapping("/{id}")
    public ResponseEntity<Survey> getSurveyById(@PathVariable String id) {
        return ResponseEntity.ok(surveyService.findSurveyById(id));
    }

    @PostMapping
    public ResponseEntity<Survey> createSurvey(@RequestBody Survey survey) {
        return ResponseEntity.ok(surveyService.createSurvey(survey));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteSurvey(@PathVariable String id) {
        surveyService.deleteSurvey(id);
        return ResponseEntity.noContent().build();
    }
}

File: project/hyunwoo-onboarding/presurvey-api/src/main/kotlin/com/innercircle/presurveyapi/api/survey/SurveyController.kt
package com.innercircle.presurveyapi.api.survey

import com.innercircle.presurveyapi.api.survey.dto.CreateSurveyRequest
import com.innercircle.presurveyapi.api.survey.dto.SurveyResponse
import com.innercircle.presurveyapi.application.survey.command.CreateSurveyUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/surveys")
class SurveyController(
    private val createSurveyUseCase: CreateSurveyUseCase
) {

    @PostMapping
    fun createSurvey(@RequestBody request: CreateSurveyRequest): ResponseEntity<SurveyResponse> {
        val command = request.toCommand()
        val survey = createSurveyUseCase.invoke(command)
        return ResponseEntity.ok(SurveyResponse.from(survey))
    }
}